lib/openhab/log/logger.rb

```diff
@@ -198,8 +192,8 @@ def log_caller
                         .grep_v(%r{openhab/core/})
```

Comment:
log_caller is a confusing name for what we are really trying to do here.  I think rules_file is more descriptive.

Reply:
An issue with this code is that the rule_file name isn't always available - like for example if a timer thread is kicked off - it isn't tied to a rules file because that isn't its base.  However, every rules file is tied to a single JRuby context, therefore if we get a rule_file name once, you can memoize it at the class level so then every log entry can have the corresponding rules file in its prefix.

Reply:
> An issue with this code is that the rule_file name isn't always available - like for example if a timer thread is kicked off - it isn't tied to a rules file because that isn't its base.

I haven't come across this scenario. Could you write an example code / cucumber test to demonstrate?

The only time I encountered where rule_file isn't available was inside scriptLoaded / scriptUnloaded, because they get invoked directly from java.

Reply:
The latest push renamed it to `rules_file`